kafka/protocol/parser.py

```diff
@@ -181,3 +190,11 @@ def _reset_buffer(self):
         self._receiving = False
         self._header.seek(0)
         self._rbuffer = None
+
+    @staticmethod
+    def is_flexible_version_request(request):
```

Comment:
I think we should move this logic to the Request class as it seems like it's primary function is to determine which version of request and response headers we should be using for a particular api+version.

Reply:
done